plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTransforms.java

```diff
@@ -703,21 +741,38 @@ static long epochHour(long epochMilli)
     public static class ColumnTransform
     {
         private final Type type;
+        private final boolean mayReturnNullForNonNull;
+        private final boolean monotonic;
         private final Function<Block, Block> blockTransform;
         private final ValueTransform valueTransform;
 
-        public ColumnTransform(Type type, Function<Block, Block> blockTransform, ValueTransform valueTransform)
+        public ColumnTransform(Type type, boolean mayReturnNullForNonNull, boolean monotonic, Function<Block, Block> blockTransform, ValueTransform valueTransform)
         {
             this.type = requireNonNull(type, "type is null");
+            this.mayReturnNullForNonNull = mayReturnNullForNonNull;
+            this.monotonic = monotonic;
             this.blockTransform = requireNonNull(blockTransform, "transform is null");
             this.valueTransform = requireNonNull(valueTransform, "valueTransform is null");
         }
 
+        /**
+         * Result type.
+         */
         public Type getType()
         {
             return type;
         }
 
+        public boolean isMayReturnNullForNonNull()
+        {
+            return mayReturnNullForNonNull;
+        }
+
+        public boolean isMonotonic()
```

Comment:
from a mathematical sense, this should be isStrictlyMonotonic (which is different than isMonotonic).

Reply:
I think that partitioning transforms are rarely (never?) strictly monotonic.